data/moves.ts

```diff
@@ -321,19 +321,26 @@ export const Moves: {[moveid: string]: MoveData} = {
 		pp: 15,
 		priority: 2,
 		flags: {metronome: 1},
-		stallingMove: true,
 		onPrepareHit(pokemon) {
-			return !!this.queue.willAct() && this.runEvent('StallMove', pokemon);
-		},
-		onTryHit(source) {
-			if (source.side.active.length === 1) return false;
-			if (source.side.active.length === 3 && source.position === 1) return false;
+			return this.runEvent('AllySwitchStallMove', pokemon);
```

Comment:
I've come up with an approach that I believe would simplify the code.
```suggestion
			return pokemon.addVolatile('allyswitch');
```
Since the Gen 8 code overrides `onPrepareHit`, this means that its `onHit` code is now the same as for Gen 9 and can therefore be removed. As for the condition, it would be inline in the move data, rather than being in `conditions.ts`:
```typescript
			condition: {
				duration: 2,
				counterMax: 729,
				onStart() {
					this.effectState.counter = 3;
				},
				onRestart(pokemon) {
					// this.effectState.counter should never be undefined here.
					// However, just in case, use 1 if it is undefined.
					const counter = this.effectState.counter || 1;
					this.debug("Ally Switch success chance: " + Math.round(100 / counter) + "%");
					const success = this.randomChance(1, counter);
					if (!success) {
						delete pokemon.volatiles['allyswitch'];
						return false;
					}
					if (this.effectState.counter < (this.effect as Condition).counterMax!) {
						this.effectState.counter *= 3;
					}
					this.effectState.duration = 2;
				},
			},
```

Reply:
This works great! Thanks urkerab!